Convert to table-driven tests and use go-check.

fernape opened the pull request:
Convert similar tests to table-driven ones.
Use [go-check](http://labix.org/gocheck) as testing framework.

This will likely affect canonical/sqlair#17, canonical/sqlair#19 and canonical/sqlair#20

fernape committed:
Convert to table-driven tests and use go-check.

Convert similar tests to table-driven ones.
Use go-check as testing framework.

niemeyer commented on internal/parse/parts_test.go:
```diff
@@ -4,30 +4,37 @@ import (
-	assert.NotEqual(t, nil, i)
-	assert.Equal(t, "mytype", i.Source.Prefix)
-	assert.Equal(t, "mytag", i.Source.Name)
+	c.Assert(i, Not(IsNil))
```
This is impossible to be nil unless the compiler is broken. Indeed, it's not even a pointer.

As an aside, IIRC we have NotNil, but typically we don't do this because we always test more details about the value than simply it being nil or not. If it was nil, the logic below will break, for instance.

fernape replied:
I think the original test didn't make sense either since this was not a pointer anyway. The rest of the tests looks good to me here and in `parser_test.go`

niemeyer commented on internal/parse/parts_test.go:
```diff
@@ -4,30 +4,37 @@ import (
-	assert.Equal(t, "mycolumn", p.Source[0].Name)
-	assert.Equal(t, "mytype", p.Target[0].Prefix)
-	assert.Equal(t, "mytag", p.Target[0].Name)
+	c.Assert(p, Not(IsNil))
```
Same here. It's worth reviewing the rest of the PR for those cases.

fernape replied:
Done.

niemeyer reviewed (changes requested):
Looks good, with only a couple of trivials. THanks for converting the tests over.

fernape committed:
Remove superfluous check in tests.

The test was translated "as is", but it doesn't make sense to check
against nil since it is not a pointer anyway.

niemeyer reviewed (approved):
Thanks for tuning, and thanks for de-conflicting, Alastair.
